src/testing/systest/supervisor/network_faults.zig

```diff
@@ -0,0 +1,173 @@
+            netem_rules.delay = null;
+            try netem_sync(allocator);
+        },
+        .network_loss_add => {
+            assert(netem_rules.loss == null);
+            const loss_pct = random.intRangeAtMost(u8, 5, 100);
+            netem_rules.loss = .{
+                .loss_pct = loss_pct,
+                .correlation_pct = 75,
+            };
+            try netem_sync(allocator);
+        },
+        .network_loss_remove => {
+            assert(netem_rules.loss != null);
+            netem_rules.loss = null;
+            try netem_sync(allocator);
+        },
+    }
+}
+
+/// Zeroes out the weights of actions that are not enabled.
+pub fn adjusted_weights(weights: arbitrary.EnumWeights(Action)) arbitrary.EnumWeights(Action) {
+    return .{
+        .network_delay_add = if (netem_rules.delay != null) 0 else weights.network_delay_add,
+        .network_delay_remove = if (netem_rules.delay == null) 0 else weights.network_delay_remove,
+        .network_loss_add = if (netem_rules.loss != null) 0 else weights.network_loss_add,
+        .network_loss_remove = if (netem_rules.loss == null) 0 else weights.network_loss_remove,
+    };
+}
+
+fn netem_sync(allocator: std.mem.Allocator) !void {
+    const args_max = (std.meta.fields(NetemRules).len + 1) * 8;
+    var args_buf = std.mem.zeroes([args_max][]const u8);
+    var args = std.ArrayListUnmanaged([]const u8).initBuffer(args_buf[0..]);
+
+    args.appendSliceAssumeCapacity(&.{ "tc", "qdisc", "replace", "dev", "lo", "root", "netem" });
+
+    var args_delay_time = std.mem.zeroes([6]u8);
+    var args_delay_jitter = std.mem.zeroes([6]u8);
+    var args_delay_correlation = std.mem.zeroes([4]u8);
+    var args_loss_pct = std.mem.zeroes([4]u8);
+    var args_loss_correlation = std.mem.zeroes([4]u8);
+
+    if (netem_rules.delay) |delay| {
+        args.appendAssumeCapacity("delay");
+        args.appendAssumeCapacity(try std.fmt.bufPrint(
+            args_delay_time[0..],
+            "{d}ms",
+            .{delay.time_ms},
+        ));
+        args.appendAssumeCapacity(try std.fmt.bufPrint(
+            args_delay_jitter[0..],
+            "{d}ms",
+            .{delay.jitter_ms},
+        ));
+        args.appendAssumeCapacity(try std.fmt.bufPrint(
+            args_delay_correlation[0..],
+            "{d}%",
+            .{delay.correlation_pct},
+        ));
+        args.appendAssumeCapacity("distribution");
+        args.appendAssumeCapacity("normal");
+    }
+
+    if (netem_rules.loss) |loss| {
+        args.appendAssumeCapacity("loss");
+        args.appendAssumeCapacity(try std.fmt.bufPrint(
+            args_loss_pct[0..],
+            "{d}%",
+            .{loss.loss_pct},
+        ));
+        args.appendAssumeCapacity(try std.fmt.bufPrint(
+            args_loss_correlation[0..],
+            "{d}%",
+            .{loss.correlation_pct},
+        ));
+    }
+
+    // Everything stack-allocated and simple up to here. Now this feels like a shame just for
```

Comment:
It feels indeed! Would be great if this were a thing:

```zig
log.info("syncing netem: {}", stdx.fmt_join("{}", " ", args.items));

// in stdx.zig
fn fmt_join(comptime item_fmt: []const u8, sep: []const u8, items: anytype)
```

I _think_ this should be possible, although I don't remember the relevant fmt APIs off the top of my head my head. Could you them up and implement this(maybe as a separate PR to stdx even)?  I remember wanting this a couple of time and chickening out of actually doing the work :-)